src/collectors/jolokia/kafka_jolokia.py

```diff
@@ -0,0 +1,592 @@
+            if (len(details) != 7):
+                self.log.error('Error processing consumer group - %s', details[0])
+                return
+
+            ###
+            #
+            # Each line contains multiple metrics. First, we construct the common base for each metric name.
+            #
+            ###
+
+            # Each metric will start with 'zookeeper.consumer_groups'
+            metric_base = 'zookeeper.consumer_groups'
+
+            # Next up is the consumer group name
+            metric_base = metric_base + '.' + details[0]
+
+            # Followed by the topic name
+            metric_base = metric_base + '.' + details[1]
+
+            # Followed by the partition number, which we preface with "partition-" for readability
+            metric_base = metric_base + '.partition-' + details[2]
+
+            ###
+            #
+            # And now for each of the actual metric names
+            #
+            ###
+
+            # 1) Consumer offet
+            metric_name = metric_base + '.consumer_offset'
+            value = details[3]
+            self.publish(metric_name, value)
+            self.log.debug('Published metric ' + metric_name + ' with value ' + str(value))
+            
+            # 2) Broker offset
+            metric_name = metric_base + '.broker_offset'
+            value = details[4]
+            self.publish(metric_name, value)
+            self.log.debug('Published metric ' + metric_name + ' with value ' + str(value))
+
+            # 3) Consumer lag (which is broker offset minus consumer offset)
+            metric_name = metric_base + '.consumer_lag'
+            value = details[5]
+            self.publish(metric_name, value)
+            self.log.debug('Published metric ' + metric_name + ' with value ' + str(value))
+
+            # 4) Owner - This column from the Zookeeper results has a string with the name of the 
+            # consumer group's owner, or the value 'none'.  We make this into a binary 0/1 to 
+            # indicate whether or not the consumer group has an owner.  
+            metric_name = metric_base + '.has_owner'
+
+            if (details[6].lower() == 'none'):
+                value = 0
+            else:
+                value = 1
+
+            self.publish(metric_name, value)    
+            self.log.debug('Published metric ' + metric_name + ' with value ' + str(value))
+
+
+    ########
+    #
+    # The clean_up() function takes the metric name and cleans it up to make it
+    # more presentable.  The default Jolokia JMX collector only does some minimal
+    # work in this regard (replacing '=' and ':" with '_" and things like that),
+    # whereas for Kafka we need to actually rearrange the ordering of the JMX
+    # keys to make the metrics fit into a hierarchy.  Therefore, we override the
+    # clean_up function to do just that.
+    #
+    # 2016-10-26 DVG
+    #
+    ########
+
+    def clean_up(self, text):
```

Comment:
But the Kafka collector in this PR has it's own clean_up so the one from the super-class is not used. The Cassandra collector is also using it's own.